hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1157,11 +1154,6 @@ protected List<HStoreFile> doCompaction(CompactionRequestImpl cr,
     }
     replaceStoreFiles(filesToCompact, sfs, true);
 
-    // This step is necessary for the correctness of BrokenStoreFileCleanerChore. It lets the
-    // CleanerChore know that compaction is done and the file can be cleaned up if compaction
-    // have failed.
-    storeEngine.resetCompactionWriter();
```

Comment:
We can leave it in the API, but the current implementation only set the 'writer' field to null. The method body becomes empty when that field is converted into a parameter. 

If we are going to keep it, we somehow need to pass a key as a parameter to abort and remove a StoreFileWriter instance from the new Set that holds them while the compactions are in progress.

Reply:
Logically we need this to keep correctness. IIRC, the problem here is that, we can only cleanup the writer instance after we successfully commit the store files to SFT, i.e, after the replaceStoreFile method. That's why we can not just simply remove the writer instance in commitWriter, otherwise there could be data loss, i.e, the BrokenStoreFileCleanerChore may delete the store files which are written just now but have not been added to the SFT yet...

Let me check again if the new implementation can solve the problem.

Reply:
Thanks for that, you know `BrokenStoreFileCleanerChore` best.

Reply:
Ah actually there is even a problem here in the current code, let me fix it...

Reply:
Fixed, at least now we do not remove the writer from the set until after commit.